capstone/fitness/src/main/java/com/google/sps/fit/Exercise.java

```diff
@@ -113,82 +117,85 @@ public Exercise build() {
   private Exercise(){}
 
   /**
-   * Returns true if average of Exercise values are better than the given Exercise's average 
-   * values in each type.
-   * Note: Uses average to ignore set count.
+   * Returns true if Exercise's values are better than or overall equal to the given Exercise's 
+   * values in each type. Uses average if values should decrement, uses summation if not. The equal
+   * to demonstrates that an exercise cannot become better in all ways.
    * 
    * @param exercise Exercise to compare to.
    * @return whether this Exercise's average values are better than the given one's.
    */
   public boolean betterThan(Exercise exercise) {
+    boolean betterThanOverall = true;
+    boolean betterThanAtLeastOnce = false;
     for(SetType type : setValues.keySet()) {
-      float srcAvg = avg(getSetValues(type));
-      float compAvg = avg(exercise.getSetValues(type));
-      if(!type.betterThan(srcAvg, compAvg)) {
-        return false;
-      }
+      boolean betterThan = betterThan(exercise, type).orElse(false);
+      betterThanAtLeastOnce |= betterThan;
+      betterThanOverall &= betterThan || equalTo(exercise, type).orElse(false);
```

Comment:
It seems like you can just early return false here if this expression becomes false?

Reply:
done